sdk/storage/blobs/inc/az_storage_blobs.h

```diff
@@ -0,0 +1,146 @@
+}
+
+AZ_NODISCARD AZ_INLINE az_result az_storage_blobs_blob_client_init(
+    az_storage_blobs_blob_client * const client,
+    az_span const uri,
+    void * const credential,
+    az_storage_blobs_blob_client_options const * const options) {
+  AZ_CONTRACT_ARG_NOT_NULL(client);
+
+  client->uri = uri;
+  // use default options if options is null. Or use customer provided one
+  if (options == NULL) {
+    client->client_options = AZ_STORAGE_BLOBS_BLOB_CLIENT_DEFAULT_OPTIONS;
+  } else {
+    client->client_options = *options;
+  }
+
+  client->pipeline = (az_http_pipeline){
+    .policies = {
+      { .pfnc_process = az_http_pipeline_policy_uniquerequestid, .data = NULL },
+      { .pfnc_process = az_http_pipeline_policy_retry, .data = &client->client_options.retry },
+      { .pfnc_process = az_http_pipeline_policy_authentication, .data = credential },
+      { .pfnc_process = az_http_pipeline_policy_logging, .data = NULL },
+      { .pfnc_process = az_http_pipeline_policy_bufferresponse, .data = NULL },
+      { .pfnc_process = az_http_pipeline_policy_distributedtracing, .data = NULL },
+      { .pfnc_process = az_http_pipeline_policy_transport, .data = NULL },
+      { .pfnc_process = NULL, .data = NULL },
+    }, 
+    };
+
+  //Currently only BlockBlobs are supported
+  client->blob_type = az_storage_blobs_blob_type_BlockBlob();
+
+  return AZ_OK;
+}
+
+/**
+ * @brief Creates a new blob
+ *
+ * @param client a storage blobs client structure
+ * @param content blob content
+ * @param options create options for blob. It can be NULL so nothing is added to http request headers
+ * @param response a pre allocated buffer where to write http response
+ * @return AZ_NODISCARD az_storage_blobs_blob_create
+ */
+AZ_NODISCARD az_result az_storage_blobs_blob_upload(
+    az_storage_blobs_blob_client * client,
+    az_span content, /* Buffer of content*/
+    az_storage_blobs_blob_upload_options * const options,
+    az_http_response * const response);
+
+// TODO: IOT customers want to be able to resume
+
+/**
+ * @brief Gets a blob
+ *
+ * @param client a storage blobs client structure
+ * @param blob_name name of blob to be retrieved
+ * @param response a pre allocated buffer where to write http response
+ * @return AZ_NODISCARD az_storage_blobs_blob_get
+ */
+AZ_NODISCARD az_result az_storage_blobs_blob_download(
+    az_storage_blobs_blob_client * client,
+    az_span_builder content,
+    az_storage_blobs_blob_download_options options,
+    az_http_response * const response);
+
+// Http header supported
+//   Range header as option for get
+//
+
+/**
+ * @brief Deletes a blob
+ *
+ * @param client a storage blobs client structure
+ * @param blob_name name of the blob to be deleted
+ * @param response a pre allocated buffer where to write http response
+ * @return AZ_NODISCARD az_storage_blobs_blob_delete
+ */
+AZ_NODISCARD az_result az_storage_blobs_blob_delete(
+    az_storage_blobs_blob_client * client,
+    az_http_response * const response);
```

Comment:
missing parameter `az_span blob_name`

Reply:
The blob name is part of the client's URL